### PR TITLE
Type hints to emu.py

### PR DESCRIPTION
**Pull request by grdddj**
Improving the quality of an important development tool

### CODE REVIEW

File: core/emu.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
 #!/usr/bin/env python3
+from __future__ import annotations
```
Comment (prusnak) [inline core/emu.py:2]: Is this required?

I don't see any annotation where we'd need postponed evaluation, or I am wrong here?

Comment (grdddj) [reply core/emu.py:2]: I am using "union pipe" two times  - `str | Path` - and without the `future` import this fails at runtime - as we have default `3.9` python, it works natively from `3.10`

Comment (prusnak) [reply core/emu.py:2]: thanks, TIL

File: core/emu.py [hunk 2]
```diff
@@ -47,13 +49,14 @@ def run_command_with_emulator(emulator, command):
...
 
-def run_emulator(emulator):
+def run_emulator(emulator: CoreEmulator) -> int:
     with emulator:
         signal.signal(signal.SIGINT, signal.SIG_IGN)
         return emulator.wait()
 
 
-def watch_emulator(emulator):
+def watch_emulator(emulator: CoreEmulator) -> Literal[0]:
```
[the remark this answers is not in the document]
Comment (grdddj) [reply core/emu.py]: Yes, makes sense. Changed in [6783e62](https://github.com/trezor/trezor-firmware/pull/2288/commits/6783e62cb5a3ca04a0a8981dd5892b659737c4fe)

**Review by matejcik — APPROVED** [review]
👍  please squash fixup and merge

**Pull request merged**